app/scripts/modules/netflix/fastProperties/modal/fastPropertyScopeBuilder.service.js

```diff
@@ -76,7 +76,7 @@ module.exports = angular
         applicationReader
           .getApplication(appName)
           .then( (application) => {
-            return application.refresh(true)  // This is a hack. Due to a race condition in the applicationReader. The clusters don't get bound to the application before tha app returns from the function. TODO: Revisit applicationReader design.
+            return application.ready()
               .then((application) => {
```

Comment:
This changes seems to work well.  Can you just remove the 'application' from the promise success function here, Since the ready() call doesn't return the application.  Otherwise looks good.